chiselFrontend/src/main/scala/chisel3/Bits.scala

```diff
@@ -1205,7 +1205,102 @@ trait SIntFactoryBase {
 
 object SInt extends SIntFactoryBase
 
-sealed trait Reset extends Element with ToBoolable
+sealed trait Reset extends Element with ToBoolable {
+  /** Casts this $coll to an [[AsyncReset]] */
+  final def asAsyncReset(): AsyncReset = macro SourceInfoWhiteboxTransform.noArg
+
+  /** @group SourceInfoTransformMacro */
+  def do_asAsyncReset(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): AsyncReset
+}
+
+object Reset {
+  def apply(): Reset = new ResetType
+}
+
+/** "Abstract" Reset Type that can be inferred to be either [[AsyncReset]] or [[Bool]]
```

Comment:
Perhaps clarify that abstract is in the sense that it is inferred in FIRRTL to the FIRRTL-concrete AsyncReset and Bool types?

Reply:
Agreed, done